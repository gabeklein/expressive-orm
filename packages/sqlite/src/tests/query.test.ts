import { Type, Time, Query, Str } from '..';
import { database } from './helpers';

class Foo extends Type {
  name = Str();
  date = Time();
  color = Str();
}

it("will insert and retrieve a Date", async () => {
  await database([Foo]);

  const now = new Date();

  await Foo.insert({
    name: "foobar",
    color: "red",
    date: now
  })

  const date = await Query.one(where => {
    const foo = where(Foo);

    where(foo.id).is(1);

    return foo.date;
  });

  // database has limited precision
  now.setMilliseconds(0);

  // asserts Date type is preserved also
  expect<Date>(date).toBeInstanceOf(Date);
  expect<Date>(date).toEqual(now);
})

it("will create count query", async () => {
  await database([Foo]);

  const query = new Query(where => {
    const foo = where(Foo);

    where(foo.color).is("red");
  });

  const qb = query.count().toString();

  expect(qb).toMatchSnapshot();
})