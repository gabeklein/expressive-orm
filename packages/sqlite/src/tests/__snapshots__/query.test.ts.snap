// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will create count query 1`] = `
	select
	  count(*)
	from
	  \`foo\`
	where
	  \`foo\`.\`color\` = 'red'
`;
